examples/sanitize.rs

```diff
@@ -0,0 +1,60 @@
+extern crate crossbeam_epoch as epoch;
+extern crate rand;
+
+use std::sync::Arc;
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed};
+use std::thread;
+
+use epoch::{Atomic, Owned};
+use rand::Rng;
+
+fn worker(a: Arc<Atomic<AtomicUsize>>) {
+    let mut rng = rand::thread_rng();
+    let mut sum = 0;
+
+    for _ in 0..1000 {
+        let guard = &epoch::pin();
+
+        let val = if rng.gen_range(0, 10) == 0 {
+            let p = a.swap(Owned::new(AtomicUsize::new(sum)), AcqRel, guard);
+            unsafe {
+                guard.defer(move || p.into_owned());
+                p.deref().load(Relaxed)
+            }
+        } else {
+            let p = a.load(Acquire, guard);
+            unsafe {
+                p.deref().fetch_add(sum, Relaxed)
+            }
+        };
+
+        sum = sum.wrapping_add(val);
+
+        if rng.gen_range(0, 2000) == 0 {
+            let a = a.clone();
+            thread::spawn(move || worker(a)).join().unwrap();
```

Comment:
I'm curious, what's the purpose of spawning threads recursively? Could the same effect be achieved by simply spawning more threads in `main` and having the loop in `worker` be tighter (no thread spawn overhead)?

Reply:
It's just to randomize the interleaving of thread spawning/exiting and atomic operations. It's a clumsy implementation, I admit.

I've changed the test now so that threads are spawned in advance and each thread first sleeps for a random amount of time, then works with the atomic for a random amount of time.